packages/vscode-dependencies-validation/src/util.ts

```diff
@@ -0,0 +1,46 @@
+import type { DiagnosticCollection, Uri } from "vscode";
+import { dirname } from "path";
+import {
+  findDependencyIssues,
+  invokeNPMCommand,
+} from "@sap-devx/npm-dependencies-validation";
+import { isEmpty } from "lodash";
+import { VscodeOutputChannel } from "./vscodeTypes";
+
+const NOT_IN_NODE_MODULES_PATTERN =
+  /^(?!.*[\\|\/]node_modules[\\|\/]).*[\\|\/].+/;
```

Comment:
this pattern may be too complex

1. it includes the parts around the `/node_modules/` (before and after)
2. it includes both types of file separators, but you can build the pattern dynamically with path.sep
   - https://nodejs.org/api/path.html#pathsep
 3. it uses negative lookahead assertion, but that a complex regexp feature, just test for `/node_modules/` in the **base** path and and flip the condition....

Reply:
path.sep does not work in regexp
Let's try it together